subprojects/core/src/main/java/org/gradle/initialization/StartParameterBuildOptions.java

```diff
@@ -60,157 +60,157 @@
         }
 
         @Override
-        public void applyTo(StartParameter settings, Origin origin) {
+        public void applyTo(StartParameterInternal settings, Origin origin) {
             settings.setOffline(true);
         }
     }
 
-    public static class RefreshDependenciesOption extends EnabledOnlyBooleanBuildOption<StartParameter> {
+    public static class RefreshDependenciesOption extends EnabledOnlyBooleanBuildOption<StartParameterInternal> {
         public RefreshDependenciesOption() {
             super(null, CommandLineOptionConfiguration.create("refresh-dependencies", "Refresh the state of dependencies."));
         }
 
         @Override
-        public void applyTo(StartParameter settings, Origin origin) {
+        public void applyTo(StartParameterInternal settings, Origin origin) {
             settings.setRefreshDependencies(true);
         }
     }
 
-    public static class DryRunOption extends EnabledOnlyBooleanBuildOption<StartParameter> {
+    public static class DryRunOption extends EnabledOnlyBooleanBuildOption<StartParameterInternal> {
         public DryRunOption() {
             super(null, CommandLineOptionConfiguration.create("dry-run", "m", "Run the builds with all task actions disabled."));
         }
 
         @Override
-        public void applyTo(StartParameter settings, Origin origin) {
+        public void applyTo(StartParameterInternal settings, Origin origin) {
             settings.setDryRun(true);
         }
     }
 
-    public static class ContinuousOption extends EnabledOnlyBooleanBuildOption<StartParameter> {
+    public static class ContinuousOption extends EnabledOnlyBooleanBuildOption<StartParameterInternal> {
         public ContinuousOption() {
             super(null, CommandLineOptionConfiguration.create("continuous", "t", "Enables continuous build. Gradle does not exit and will re-execute tasks when task file inputs change.").incubating());
         }
 
         @Override
-        public void applyTo(StartParameter settings, Origin origin) {
+        public void applyTo(StartParameterInternal settings, Origin origin) {
             settings.setContinuous(true);
         }
     }
 
-    public static class NoProjectDependenciesRebuildOption extends EnabledOnlyBooleanBuildOption<StartParameter> {
-        public static final String DEPRECATION_MESSAGE = "Support for --no-rebuild and -a was deprecated and is scheduled to be removed in Gradle 5.0.";
+    public static class NoProjectDependenciesRebuildOption extends EnabledOnlyBooleanBuildOption<StartParameterInternal> {
         public NoProjectDependenciesRebuildOption() {
-            super(null, CommandLineOptionConfiguration.create("no-rebuild", "a", "Do not rebuild project dependencies.").deprecated(DEPRECATION_MESSAGE));
+            super(null, CommandLineOptionConfiguration.create("no-rebuild", "a", "Do not rebuild project dependencies.").deprecated());
```

Comment:
:o: See comment above about breaking out a `LONG_OPTION` and `SHORT_OPTION`.

Reply:
Done.